src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -2252,20 +2320,47 @@ private static async Task ExecuteResultWriteResponse(IResult? result, HttpContex
         await EnsureRequestResultNotNull(result).ExecuteAsync(httpContext);
     }
 
-    private static Task WriteJsonResponse(HttpResponse response, object? value, JsonSerializerOptions? options)
+    // This method will not check for polymorphism and will
+    // leverage the STJ polymorphism support.
+    private static Task WriteJsonResponseFast<T>(HttpResponse response, T value, JsonTypeInfo jsonTypeInfo)
```

Comment:
There should be no cast here.

Reply:
Yeah! The cast will be needed because `options.GetTypeInfo()` returns untyped  `JsonTypeInfo` but I believe I can move it to avoid it for every request. Let me check.